shell/platform/darwin/ios/framework/Source/FlutterObservatoryPublisher.mm

```diff
@@ -219,27 +221,32 @@ - (instancetype)init {
   self = [super init];
   NSAssert(self, @"Super must not return null on init.");
 
-  if (@available(iOS 9.3, *)) {
-    _delegate.reset([[ObservatoryDNSServiceDelegate alloc] initWithOwner:self]);
+  auto command_line = flutter::CommandLineFromNSProcessInfo();
```

Comment:
Do you think it would make more sense to have this one level higher?  Never create a FlutterObservatoryPublisher if observatory publishing is disabled instead of making one that does nothing?

Reply:
I just realized the publisher delegates are backing `-[FlutterEngine observatoryUrl]` and on this patch it's always nil.  I think that unfortunately means I have to push this check one level lower so the URL is set, but the mDNS part is skipped.  I'll fix it up.